### PR TITLE
Backport to fix #2273 in 2.x stable

### LINKED ISSUE #2273: TinyMCE editor is not loading menu based on config yml
### Descriptive summary

Hyrax 1.0.5 - The tinymce editor loads what appears to be a default menu set, ignoring menu elements included in config yml

### Rationale

Menu items to include image attachment and linking functions are missing without config yml in force.

### Expected behavior

Tinymce editor should load with menu items from config yml

### Actual behavior

Tinymce editor loads with what appears to be a default menu set, does not include menu items required by config yml

### Related work

Local fix from Scholar@UC - https://github.com/uclibs/scholar_uc/pull/1763

### PR DESCRIPTION
**Pull request by anusharanganathan**
refs https://github.com/samvera/hyrax/issues/2273

Updates tinymce-rails gem and fixes broken config loading.

This is a backport of PR https://github.com/samvera/hyrax/pull/4759 for Hyrax 2.x stable

This updates the TinyMCE gem that has been pinned at an older version for several years. It also was not loading the application's tinymce config file, instead using defaults. This PR addresses both of those issues.

Changes proposed in this pull request:
* Remove version pin on tinymce-rails gem
* Modify how its JS is loaded according to the gem's install docs.
* Update the provided config file.

Guidance for testing, such as acceptance criteria or new user interface behaviors:
* The TinyMCE editor loads without error and functions when editing Pages and Content Blocks in the dashboard.

@samvera/hyrax-code-reviewers

### CODE REVIEW

**Comment by no-reply** [comment]
this seems okay.

i was worried about backward compatibility, but it seems like there's no impact to existing apps? 

looks like we could release as 2.9.7(?)

cc: @dlpierce

File: .dassie/config/tinymce.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  <<: *default
+  menubar: false
+  toolbar1: styleselect | bold italic | undo redo
+  toolbar2: table | fullscreen | image
+  plugins:
+    - table
+    - fullscreen
+    - image
+custom:
+  <<: *default
```
Comment (dlpierce) [inline .dassie/config/tinymce.yml:13]: This is for dassie which is not in 2.x, so this file can be removed from this PR.

**Review by dlpierce — CHANGES_REQUESTED** [review]
Aside from the dassie config file, looks good and works locally for me. Thanks for the backport!

File: .dassie/config/tinymce.yml (hunk 1 above)
Comment (no-reply) [reply .dassie/config/tinymce.yml:13]: dassie was backported to 2.x sometime in the last few months!

**Review by no-reply — APPROVED** [review]

**Comment by dlpierce** [comment]
Oh, I was not aware and the .dassie directory looked empty.

**Pull request merged**